### PR TITLE
[BD-18] Standardize the make test targets

### PR DESCRIPTION
**Pull request by emzubair**
## Standardize the make test targets
Added target for test & clean

**Packages added**

- pytest-cov
- edx-lint

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @emzubair! I've created [BLENDED-458](https://openedx.atlassian.net/browse/BLENDED-458) to keep track of it in Jira.  More details are on the [BD-18](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1593934815/BD-18+Testing+Consistency+Improvements) project page.

When this pull request is ready, tag your edX technical lead.

File: Makefile [hunk 1]
```diff
@@ -20,6 +20,27 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
...
 
 requirements: ## install development environment requirements
 	pip install -qr requirements/base.txt
+	pip install -qr requirements/testing.txt
+
+quality-js: ## Run JavaScript linter
+	echo "Tool not configured for quality target"
+
+quality-python: ## Run python linters
+	echo "Tool not configured for quality target"
```
Comment (feanil) [inline Makefile:33]: Add edx-lint to this repo.

File: Makefile [hunk 2]
```diff
@@ -20,6 +20,27 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
...
 	pip-compile --upgrade -o requirements/base.txt requirements/base.in
 	pip-compile --upgrade -o requirements/server.txt requirements/server.in
+	pip-compile --upgrade -o requirements/testing.txt requirements/testing.in
 
 requirements: ## install development environment requirements
 	pip install -qr requirements/base.txt
+	pip install -qr requirements/testing.txt
+
+quality-js: ## Run JavaScript linter
+	echo "Tool not configured for quality target"
```
Comment (feanil) [inline Makefile:30]: Remove the JS target if there is no testing setup.

File: Makefile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-.PHONY: help clean upgrade requirements
+.PHONY: help clean upgrade requirements test quality
```
Comment (feanil) [inline Makefile:1]: Add the rest of the new targets to `.PHONY`

File: package-lock.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@edx/mockprock",
-  "version": "1.0.1",
+  "version": "1.0.2",
```
Comment (feanil) [inline package-lock.json:3]: Update the version in [setup.py](https://github.com/edx/mockprock/blob/master/setup.py#L41) to match this as well.

**Pull request merged**